src/NuGet.Clients/NuGet.PackageManagement.UI/PackageItemLoader.cs

```diff
@@ -192,6 +196,36 @@ public async Task UpdateStateAsync(IProgress<IItemLoaderState> progress, Cancell
             NuGetEventTrigger.Instance.TriggerEvent(NuGetEvent.PackageLoadEnd);
         }
 
+        private async Task<SearchResult<IPackageSearchMetadata>> CombineSearchAsync(
+            CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            // get the browse/search results
+            SearchResult<IPackageSearchMetadata> browseResult = await _packageFeed.SearchAsync(_searchText, SearchFilter, cancellationToken);
+
+            // get the recommender results
+            SearchResult<IPackageSearchMetadata> recommenderResult = null;
```

Comment:
nit: In the future, these should probably happen in parallel.

Reply:
Specifically serach and recommender query...a new issues might be good for that.

Reply:
Created issue https://github.com/NuGet/Home/issues/9572